code/modules/cargo/bounty_console.dm

```diff
@@ -32,65 +32,35 @@
 	if(!GLOB.bounties_list.len)
 		setup_bounties()
-
-	var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
-	var/list/dat = list({"<a href='?src=[REF(src)];refresh=1'>Refresh</a>
-	<a href='?src=[REF(src)];refresh=1;choice=Print'>Print Paper</a>
-	<p>Credits: <b>[D.account_balance]</b></p>
-	<table style="text-align:center;" border="1" cellspacing="0" width="100%">
-	<tr><th>Name</th><th>Description</th><th>Reward</th><th>Completion</th><th>Status</th></tr>"})
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "CargoBountyConsole", name, 750, 600, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/bounty/ui_data(mob/user)
+	var/list/data = list()
+	var/list/bountyinfo = list()
 	for(var/datum/bounty/B in GLOB.bounties_list)
-		if(B.claimed)
-			dat += "<tr style='background-color:#294675;'>"
-		else if(B.can_claim())
-			dat += "<tr style='background-color:#4F7529;'>"
-		else
-			dat += "<tr style='background-color:#990000;'>"
+		bountyinfo += list(list("name" = B.name, "description" = B.description, "reward_string" = B.reward_string(), "completion_string" = B.completion_string() , "claimed" = B.claimed, "can_claim" = B.can_claim(), "priority" = B.high_priority, "bounty_ref" = REF(B)))
+	data["stored_cash"] = cargocash.account_balance
+	data["bountydata"] = bountyinfo
+	return data
 
-		if(B.high_priority)
-			dat += {"<td><b>[B.name]</b></td>
-			<td><b>High Priority:</b> [B.description]</td>
-			<td><b>[B.reward_string()]</b></td>"}
-		else
-			dat += {"<td>[B.name]</td>
-			<td>[B.description]</td>
-			<td>[B.reward_string()]</td>"}
-		dat += "<td>[B.completion_string()]</td>"
-		if(B.claimed)
-			dat += "<td>Claimed</td>"
-		else if(B.can_claim())
-			dat += "<td><A href='?src=[REF(src)];refresh=1;choice=Claim;d_rec=[REF(B)]'>Claim</a></td>"
-		else
-			dat += "<td>Unclaimed</td>"
-		dat += "</tr>"
-	dat += "</table>"
-	dat = dat.Join()
-	var/datum/browser/popup = new(user, "bounties", "Nanotrasen Bounties", 700, 600)
-	popup.set_content(dat)
-	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
-	popup.open()
-
-/obj/machinery/computer/bounty/Topic(href, href_list)
+/obj/machinery/computer/bounty/ui_act(action,params)
 	if(..())
 		return
-
-	switch(href_list["choice"])
+	var/datum/bounty/cashmoney = locate(params["bounty"]) in GLOB.bounties_list
+	switch(action)
+		if("ClaimBounty")
+			if(cashmoney)
+				cashmoney.claim()
+			return
 		if("Print")
 			if(printer_ready < world.time)
 				printer_ready = world.time + PRINTER_TIMEOUT
 				print_paper()
-
-		if("Claim")
-			var/datum/bounty/B = locate(href_list["d_rec"]) in GLOB.bounties_list
-			if(B)
-				B.claim()
-
-	if(href_list["refresh"])
-		playsound(src, "terminal_type", 25, FALSE)
-
-	updateUsrDialog()
+				return
+	. = TRUE

```

Comment:
Why are you updating if there was no changes. This should only return true for claim here from what i can see.